### ISSUE TITLE
Error in help message; says "terraform providers lock -provider=darwin_amd64"

### ISSUE THREAD

**Issue by jeliker** [hashicorp/terraform#31709]
### Terraform Version

```shell
$ terraform -v
Terraform v1.2.8
on darwin_amd64
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/oci v4.89.0
+ provider registry.terraform.io/hashicorp/template v2.2.0
```

### Terraform Configuration Files

N/A

### Debug Output

```
$ terraform init

Error: Failed to install provider
│ 
│ Error while installing hashicorp/oci v4.89.0: the current package for registry.terraform.io/hashicorp/oci 4.89.0 doesn't match any of the checksums previously recorded in the dependency
│ lock file
│ 
│ You can ensure the current platform, darwin_amd64, is included in the dependency lock file by running: `terraform providers lock -provider=darwin_amd64`.
│ 
│ If this does not fix the problem you may need to reset any provider caching present in your setup or make sure you are connecting to valid provider distributions.
```

### Expected Behavior

Error message example should show

> terraform providers lock **-platform**=darwin_amd64

### Actual Behavior

Error message shows (`-provider` not `-platform`)

> terraform providers lock -provider=darwin_amd64

### Steps to Reproduce

1. `terraform init` when provider was updated on another system, another platform

**Comment by crw** [comment]
Thanks for this report!

**Comment by kmoe** [comment]
This is fixed in v1.3.0 and v1.2.9 - please reopen this issue if you believe the bug still exists in these versions.
